clientv3/balancer.go

```diff
@@ -140,27 +143,23 @@ func (b *simpleBalancer) Up(addr grpc.Address) func(error) {
 		return func(err error) {}
 	}
 
-	if len(b.upEps) == 0 {
+	if b.pinAddr == "" {
 		// notify waiting Get()s and pin first connected address
-		close(b.upc)
 		b.pinAddr = addr.Addr
+		close(b.upc)
```

Comment:
it doesn't matter since it's protected by a lock so it's atomic, but minimum diffs are generally preferable...